### ISSUE TITLE
macOS Apple Silicon build is here.

### ISSUE THREAD

**Issue by cxy5014** [g4ixt/QtTinySA#41]
I have built this project for Apple Silicon based macOS. You can find the download link below.

https://github.com/cxy5014/QtTinySA/releases/tag/v0.10.1

**Comment by g4ixt** [comment]
Thank you.  Do you intend to keep building executables for Mac when I release updates?  If so I will put a link to your repo on the Wiki.

Because I can't test or verify Mac executables I am not going to put them in my 'release' folder.

Ian

**Comment by cxy5014** [comment]
Sure, I will continue building executables for Mac once you release updates. Next, I'm going to try GitHub Actions for automatic builds.

Bill BG5VGK

**Comment by g4ixt** [comment]
I have added a Wiki page for it.  https://github.com/g4ixt/QtTinySA/wiki

**Comment by cxy5014** [comment]
The Mac Executables is now built by Github Actions.

**Comment by g4ixt** [comment]
Bill,

Could you try out my development branch version v0.11.1 and tell me if the fix for Windows is relevant to macOS?  It's in this function which is currently around line 131.  Essentially what I need to know is does macOS return port.product as "tinySA" or "tinySA4" from pySerial list_ports or does it return 'None' like Windows.  If it returns the correct value I can modify the code so macOS works like Linux, which will be better.

    def identify(self, port):
        # Windows returns no information to pySerial list_ports.comports()
        if system() == 'Linux':
            return port.product
        else:
            return 'USB device'

**Comment by cxy5014** [comment]
Here is the ScreenShot.
<img width="1112" alt="Screenshot 2024-08-11 at 14 49 42" src="https://github.com/user-attachments/assets/31b4cc64-63c9-415c-a62c-73bd6031911b">

**Comment by g4ixt** [comment]
The programme looks really nice in macOS, I have never seen a screenshot before; thank you, I will add it to the screenshots in the repository.  I must have a read about what Github actions do.

I wasn't very clear in my request, could you change the line "if system() == 'Linux':" to be
"if system() == 'Linux' or system() == 'Darwin':

(Google search tells me 'Darwin' is what macOS returns for platform.system)

Also check that the new combobox in the preferences screen is populating correctly.

Thanks again

**Comment by cxy5014** [comment]
Hi,
I have just confirmed that changing the line "if system() == 'Linux':" to be "if system() == 'Linux' or system() == 'Darwin':" is the same with tinySA4 just like the screenshot I have sent before. The combobox run correctly. The screen recorded is attached.
![Screen Recording 2024-08-11 at 19 51 40-2](https://github.com/user-attachments/assets/250c0d43-5631-4460-a0c5-dec365952061)

**Comment by g4ixt** [comment]
Excellent.  Presumably the 3 coloured dots top left are the file, preferences, and help menus

**Comment by cxy5014** [comment]
Nope, the red dot is close, yellow one is minimize, green one is full screen. 
<img width="545" alt="Screenshot 2024-08-12 at 17 16 25" src="https://github.com/user-attachments/assets/22571fc4-a25b-4063-81a4-13458f789e79">

**Comment by cxy5014** [comment]
Besides, the database 'QtTSAprefs.db' file path is /User/username/Library/Application Support/QtTinySA

**Comment by g4ixt** [comment]
> Besides, the database 'QtTSAprefs.db' file path is /Users/username/Library/Application Support/QtTinySA

I accidentally lost a line of code that affected the database location.  I have restored it in the version I am working on.
